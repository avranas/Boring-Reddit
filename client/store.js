import { configureStore } from "@reduxjs/toolkit";
import searchResultsReducer from "./slices/searchResultsSlice";
import currentPostReducer from "./slices/currentPostSlice";
import optionsReducer from "./slices/optionsSlice";

export default configureStore({
  reducer: {
    searchResults: searchResultsReducer,
    currentPost: currentPostReducer,
    options: optionsReducer,
  },
});
