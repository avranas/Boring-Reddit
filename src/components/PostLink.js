import { Link } from 'react-router-dom';
import updoot from '../images/updoot.png';

const PostLink = (props) => {

  return (
      <tr key={props.title}>
        <td class="upvotes">
          <img alt={"upvote"} src={updoot} id='upvote-img'/>
          </td>
        <td class="upvotes">
          <p>{props.ups}</p>
        </td>
        <td>
          <tr>
            <td>
              <Link to={`/post?redditUrl='${props.redditUrl}'`}>{props.title}</Link>
            </td>
          </tr>
          <tr>
            <td>
              <p>u/{props.author}</p>
            </td>
          </tr>
        </td>
      </tr>
  );
}

export default PostLink;
