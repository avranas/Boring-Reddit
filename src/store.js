import { configureStore } from "@reduxjs/toolkit";
import searchResultsReducer from "../src/slices/searchResultsSlice";
import currentPostReducer from "../src/slices/currentPostSlice";
import optionsReducer from "../src/slices/optionsSlice";

export default configureStore({
  reducer: {
    searchResults: searchResultsReducer,
    currentPost: currentPostReducer,
    options: optionsReducer,
  },
});
