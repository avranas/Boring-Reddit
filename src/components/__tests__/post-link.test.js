import { render, screen, cleanup } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import PostLink from '../PostLink';
import { Provider } from 'react-redux';
import store from '../../store';
import renderer from 'react-test-renderer';

afterEach(() => {
  cleanup();
});

//I will need to replace this with another reddit post if this gets removed.
const props = {
}

const postLink = 
  <Provider store={store}>
    <BrowserRouter>
      <table>
        <tbody>
          <PostLink 
            key={1}
            id={1}
            title={"This is a post on Reddit"}
            redditUrl={'https://www.reddit.com/r/tifu/comments/x35iu6/tifu_my_20f_girlfriend_of_two_years_told_me_the/'}
            nsfw={false}
            author={'TylerLife'}
            ups={170000}
          />
        </tbody>
      </table>
    </BrowserRouter>
  </Provider>;

test('should render PostLink component', () => {
  render(postLink);

    //TODO: WHY?
  const postLinkElement = screen.getByTestId('post-link-1');
  expect(postLinkElement).toBeInTheDocument();
  expect(postLinkElement).toHaveTextContent(170000);
  expect(postLinkElement).toHaveTextContent('TylerLife');
})

test('PostLink matches snapshot', () => {
  const tree = renderer.create(postLink).toJSON();
  expect(tree).toMatchSnapshot();
});
