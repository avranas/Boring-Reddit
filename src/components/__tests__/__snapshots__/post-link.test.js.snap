// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostLink matches snapshot 1`] = `
<table>
  <tbody>
    <tr
      data-testid="post-link-1"
    >
      <td
        className="upvotes"
      >
        <img
          alt="upvote"
          id="upvote-img"
          src="updoot.png"
        />
      </td>
      <td
        className="upvotes"
      >
        <p>
          170000
        </p>
      </td>
      <td>
        <table>
          <tbody>
            <tr>
              <td>
                <a
                  href="/post?redditUrl='https://www.reddit.com/r/tifu/comments/x35iu6/tifu_my_20f_girlfriend_of_two_years_told_me_the/'"
                  onClick={[Function]}
                >
                  This is a post on Reddit
                </a>
              </td>
            </tr>
            <tr>
              <td>
                <p
                  className="author"
                >
                  u/
                  TylerLife
                </p>
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;
